### ISSUE TITLE
Landis Shakedown tests part 2

### ISSUE THREAD

**Issue by DocCyblade** [DocCyblade/tkl-odoo#55]
Going to use this to document my shake down testing of RC1 part 2

Test will be done with VMWare Esxi with
2 X CPU
4GB RAM
20GB HDD

Test will be done with DEMO data 

Test one will install all modules available and test each one using CRUD and every option at least 3 times. 

Goal is to test all modules and options with limited resources. Will note any issues here. 

Retest anything that failed due to resources again here 

 Once testing is done, add two more databases with all modules, for a total of three databases. Report on any issues with multi databases and the functionalities of them

Keep list of things tested here only. Add comments if any issues only.

Test List:
- [ ] ISO Live Preview (Basic new database and module) **Fails** with 512MB RAM would need 1GB
- [x] ISO Install Guided Setup all
- [ ] ISO Install Guided Setup LVM (Defaults)
- [ ] First Boot: Signup for email alerts
- [x] First Boot: Updates
- [x] Create new database
- [x] Update Company Info
- [x] Add two users
- [x] Delete user
- [x] Install and configure all local APPS
        LIST BELOW ARE "APPS".  
- [ ] Install and configure some local Modules
        Modules gets Big.  APPS are CORE and GOOD TEST.
        Can See what Modules have been installed by Choosing Dropdown Triangle in search box
        I count that 60 MODULES have been installed alongside Base Apps just w/ App installs, no  
        Module Installs themselves.
- [ ] Test APPS -  (Basic function of each)
- [ ] Test Some Extra Modules- then  (Basic function of each)
- [x] CRM
- [x] Social Network
- [x] Online Billing
- [x] Point of Sale
- [x] Project Management
- [x] Notes
- [x] Issue Tracking
- [x] Accounting and Finance
- [x] Survey
- [x] Sales Management
- [x] Warehouse Management
- [x] Instant Messaging
- [x] MRP
- [x] Purchase Management
- [x] Employee Directory
- [x] Timesheets
- [x] Recruitment Process
- [x] Leave Management
- [x] Expense Tracker
- [x] Employee Appraisals
- [x] Calendar
- [x] Address Book
- [x] Gamification
- [x] Live Chat
- [x] Odoo Live Support
- [x] Lunch Orders
- [x] Website Builder
- [x] Blogs
- [x] Online Events
- [x] Forum
- [x] eCommerce
- [x] Fleet Management

**Comment by l-arnold** [comment]
Building off of Dev as it looks inline with RC2 right now and I have a folder built already.
make clean && make
running now.

**Comment by l-arnold** [comment]
It should run this find as I run "nearly the same" with 2 gb...  ok a bit more above but no JobRunner or Magento-Connector.

**Comment by DocCyblade** [comment]
Also you can do a `git checkout v14-RC1` after doing a `git fetch`

**Comment by l-arnold** [comment]
Whups, Me Too, Just posted in the wrong issue.

Think I should update these or are we good?

limit_memory_soft = 671088640
limit_memory_hard = 1610612736

**Comment by l-arnold** [comment]
OK,  I see this is all sequenced.  
Will start with #53

**Comment by l-arnold** [comment]
Ok. My Methodology will be to Create some critical Snapshots.

Install Reboot
Sales Management

then I will put some big Modules in and maybe go back after to a different big module.

Will see if I can get 3 Gb. Maybe 4 if I turn a few VM's off for a bit.

**Comment by l-arnold** [comment]
Processor Usage Peaks during install.  Memory usage is not really going about 512mb  (2 workers at 256 I guess is the answer there)..  but not clear yet.  Will keep plugging.

**Comment by l-arnold** [comment]
Website working.  Seems we could consider increasing the "soft" memory limits.  I don't seem to exceed about 569 mb even though 4 Gb available. 
My understanding of HTOP is 560/4000 would indicate the above statement.
Processors will peak frequently at 100% and seems to be somewhat of a bottleneck at times.

**Comment by l-arnold** [comment]
Increased Soft Limit to 1GB and restarted Service.  Don't see much difference in HTOP however.

**Comment by l-arnold** [comment]
Basic Order processed on Website (seems to be in the "cash" mode which makes sense for a Demo.

**Comment by l-arnold** [comment]
Order Rendered on Back End.  Have not set the Admin Email address yet but rewrote to my own.

No Email on incoming yet.  Probably need to configure some things there first.

**Comment by l-arnold** [comment]
made Invoice.  Went to process payment:  Odoo Error:

ValidateError

Field(s) `currency_id` failed against a constraint: The selected account of your Journal Entry forces to provide a secondary currency. You should remove the secondary currency on the account or select a multi-currency view on the journal.

**Comment by l-arnold** [comment]
Point of Sale Module works.

**Comment by l-arnold** [comment]
Lunch Orders works.  I need that one here.

**Comment by l-arnold** [comment]
Manufacturing,  Human Resources, Fleet Management, Purchases, and Project all seem to be working  in Demo.

**Comment by l-arnold** [comment]
Run of Schedulers in "Warehous Management has moved Ram Usage up to 647 mb (of 4 gb)

Pretty level for now 647-648 mb...  w/ HTOP

Processor usage low.

**Comment by l-arnold** [comment]
Definitely pretty fast with 4 gb.  Will dial this build down to 2 Gb ram and see if differences.

**Comment by l-arnold** [comment]
Added WMS Landed Cost
Added Payroll
Added Asset Management

Correction : Ram now at 760mb / of 4 gb)

**Comment by l-arnold** [comment]
Google Calendar Module gave an Error on installation

Warning, Each Constraint Module must be unique

(shows installed,  Ram usage goes to 825 mb)

**Comment by l-arnold** [comment]
Google Spreadsheet lost its icon.  Lots of new Country Accounting packages have shown up in possible installs..  This goes on and on.

Pushing 900 mb after Google Spreadsheet

Membership Association 914 mb, then 948mb after something.....
Partner Assignation and Geolocation added a bit ago as well.

**Comment by l-arnold** [comment]
Multiple Analytic Plans got me up to 999 mb / 4gb avail

**Comment by l-arnold** [comment]
Restaurant Module adds 5-6mb to Memory Load.
1005 mb total

Can't find what the Restaurant Module actually did however.

**Comment by l-arnold** [comment]
Claims Managemetn 1012 mb (so 7 mb extra)..  peaked at 13 mb extra during install.

**Comment by l-arnold** [comment]
Chat seems to work.  (need to log in on another machine and not Call Odoo)

**Comment by l-arnold** [comment]
Seems all good at 4 GB.
Ram usage does not really change with 1 user logging in or out.
1015 mb Max usage right now.

**Comment by l-arnold** [comment]
Memory Limit has been set at:
limit_memory_soft =  1073741824
limit_memory_hard = 3221225472

Dialing down next  to:
limit_memory_soft =  268435456  256mb
limit_memory_hard = 1610612736  1.5 gb

**Comment by l-arnold** [comment]
openerp-server.servic  Restart shows 247 in HTOP immediately.

4gb still available.  Will Shutdown and move RAM to 2 gb.

**Comment by DocCyblade** [comment]
@l-arnold - If you find something you want me to investigate or that bugs you in your testing, create a new issue tag it with bug and v14-READY.  Check first here to see if I already marked it [LIST BUGS](https://github.com/DocCyblade/tkl-odoo/issues?utf8=✓&q=is%3Aopen+milestone%3Av14-READY+label%3Abug+)

Also when you get a chance see what the disk usage is for the postgres database

`du -sh /var/lib/postgresql/9.4/main`

**Comment by l-arnold** [comment]
Lost my comment.  Moved to 2 GB Build.
Using at start 511 mb of 2 gb

(Noted also that 4 Gig Setup used 4 Gig Swap File which fundamentally made a 15.x GB Hard Drive.
Using 17 % (or so I recall) of the HD w/ no real data but all modules loaded.

Using 25% of the 2 GB Ram.  Will run some sessions.

POS takes a bit to start up Processor wise but actually reduced RAM consumption.

**Comment by l-arnold** [comment]
Slower Response Noted after exiting Point of Sale to jump between modules at 2 gb.
Running 1.5 gb max and 256mb in openerp-server.conf memory limits settings.

May want to bump to 512mb for the solft limit instead of 256mb.

**Comment by l-arnold** [comment]
Faster now with 512mb Soft Limit.
Renders an Order to both Email and Print.

Strangely HTOP is showing less memory consumption: 447 mb now.

**Comment by l-arnold** [comment]
GeoLocalize Module is nice.  Need to see if I can render locations to Maps as well within ODOO.

**Comment by l-arnold** [comment]
Second Database before modules installed adds about   100mb Load to Ram Consumption running on a 2 gb w/ 1.5gb HargLimit and 512mb Soft Limit

An extra 50 mb once CRM Module is loaded so 100 MB so far.  (yet another 50 (630) so 150 add for second database.

One Rule.  Don't keep unused Databases.  Would take a lot of energy to keep all of them active and once you have 2, the Web System starts to break down.  Takes special Config to make multiple Databases live and responsive to diff hosts.  (don't know how just now)

UPDATE:  With more than One, then everyone needs to log in before seeing anything.

**Comment by l-arnold** [comment]
Sales Management adds an additional 45 mb -> 674mb used.

**Comment by l-arnold** [comment]
New Login to Shell,  Indicates 
44% of Memory being used.
.54 System Load
12.7% of HD
126 processes
No SWAP

**Comment by l-arnold** [comment]
711 mb of 2 GB Ram being used after POS Installed 
713 mb when started as a session

No Products to sell somehow... however.

Forgot to load DEMO DATA

**Comment by l-arnold** [comment]
3rd Database Loaded, this time with Demo DATA.
Ram Jumps to 760 mb used (of 2 gb)

CRM Loaded:  Jumps to 782 gb

All Marketing Modules Loaded:  Brings RAM UP to 791 mb usage.

**Comment by l-arnold** [comment]
Sales Management loaded:
830 mb
now
869 mb after full install.

**Comment by l-arnold** [comment]
Several app modules get loaded automatically (need to study which do what)
Wharehouse Management
Purchasing Mangement
Instant Messageing to name some APPS

Several Modules also as noted before
Now 875 mb usage.

**Comment by l-arnold** [comment]
add Notes 909 mb
add Issue Tracking 936 mb
Accounting and Finance (during install) 953mb
Point of Sale 960 mb

**Comment by l-arnold** [comment]
Dropped TEST2 Database (empty - no demo data)
910 mb
(50 mb to place hold a database)

**Comment by l-arnold** [comment]
Live Chat Installed:  now 963  (also 50 mb or so)

**Comment by l-arnold** [comment]
Strangely, installing WEBSITE builder did not increase RAM consumption at all.

I think that is because it renders easier than the Odoo Client Interface.

**Comment by l-arnold** [comment]
Blog Posts plus MRP jumps memory to 1046 memory

This is on 2 x databases installed at present.

**Comment by l-arnold** [comment]
I think this is generally telling.  Let me know if anything should be checked now.

**Comment by l-arnold** [comment]
Google Calendar Installed with NO ERRORS on database 2 (test2)

**Comment by l-arnold** [comment]
278 Modules and Apps total in Base Install

**Comment by l-arnold** [comment]
No Information in Settings on how to implement "Website Google Map".

Would be nice to have a back end Mapping System to find customer locations.

**Comment by l-arnold** [comment]
Seems we can close this.  4 Gigs seems comfortable until we might get lots of users or Web Hits.

**Comment by l-arnold** [comment]
Wrong Issue Closed.

**Comment by l-arnold** [comment]
Nope,  Correct Issue closed.

**Comment by l-arnold** [comment]
Now it thinks they are all installed however..  
Oh well, just testing.

An Odoo thing.
